packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/indices.ts

```diff
@@ -4,61 +4,49 @@ import {
   GraphObjectIteratee,
   IntegrationError,
 } from '@jupiterone/integration-sdk-core';
-import { FlushedEntityData, FlushedRelationshipData } from '../types';
-import {
-  readJsonFromPath,
-  walkDirectory,
-  WalkDirectoryIterateeInput,
-} from '../../fileSystem';
+import { readJsonFromPath, WalkDirectoryIterateeInput } from '../../fileSystem';
 
 import { buildIndexDirectoryPath } from './path';
+import { iterateParsedGraphFiles } from '../..';
 
 interface IterateIndexInput<GraphObject> {
   type: string;
   iteratee: GraphObjectIteratee<GraphObject>;
 }
 
-export async function iterateEntityTypeIndex<T extends Entity = Entity>({
+interface BaseIterateIndexInput<GraphObject>
+  extends IterateIndexInput<GraphObject> {
+  collectionType: 'entities' | 'relationships';
+}
+
+async function iterateTypeIndex<T extends Entity | Relationship>({
   type,
+  collectionType,
   iteratee,
-}: IterateIndexInput<T>) {
+}: BaseIterateIndexInput<T>) {
   const path = buildIndexDirectoryPath({
-    collectionType: 'entities',
+    collectionType,
     type,
   });
 
-  await walkDirectory({
-    path,
-    iteratee: async (input) => {
-      const object = await readGraphObjectFile<FlushedEntityData>(input);
-      if (isObjectFlushedEntityData(object)) {
-        for (const entity of object.entities as T[]) {
-          await iteratee(entity);
-        }
-      }
-    },
-  });
+  await iterateParsedGraphFiles(async (data) => {
+    for (const graphObj of (data[collectionType] as T[]) || []) {
+      await iteratee(graphObj);
+    }
+  }, path);
```

Comment:
FWIW I find it a bit more readable when the callback is the last argument, rather than the first, but not a big deal.

Reply:
Agreed! Good feedback. This code was here previously, so I didn't change the signature.